Tickets/dm 30844

pcortes80 opened the pull request:
Please check the code after all changed made. Thanks.

teweitsai commented on doc/detailsDefault.md:
```diff
@@ -0,0 +1,47 @@
+LabVIEW Component will read this file to register commands, events and telemetry in TCP Server.
+
+The **default.ini** file is in `config/` directory.
+You can follow [here](../config/default.ini) for details.
```
This (../config/default.ini) is the `ini` file instead of the details.

pcortes80 replied:
the idea is click and go to the ini file, so I just modified the text.

teweitsai commented on doc/detailsDefault.md:
```diff
@@ -0,0 +1,47 @@
+### events
+
+1. **compName** : Specify which component's event is being subscribed.
+2. **id** : Specify the name of event that begins with `evt_`.
```
In the `ini` file, the evt_` is not needed. Right?

pcortes80 replied:
You are right, I will remove the `evt_`.

teweitsai commented on doc/detailsDefault.md:
```diff
@@ -0,0 +1,47 @@
+### telemetry
+
+1. **compName** : Specify which component's telemetry is being subscribed.
+2. **id**: Specify the name of telemetry that begins with `tel_`.
```
In the `ini` file, the `tel_` is not needed. Right?

pcortes80 replied:
You are right, I will remove the `tel_ `.

teweitsai commented on doc/detailsDefault.md:
```diff
@@ -0,0 +1,47 @@
+| compName |          id         |
+|:--------:|:-------------------:|
+|  MTMount | elevationInPosition |
+|   M1M3   |  azimuthInPosition  |
```
M1M3 has no `azimuthInPosition `. Right?

pcortes80 replied:
Modified M1M3 by MTMount.

teweitsai commented on doc/detailsDefault.md:
````diff
@@ -0,0 +1,47 @@
+
+```text
+[events]
+compName=MTMount,M1M3
````
M1M3 has no `azimuthInPosition `. Right?

pcortes80 replied:
Modified M1M3 by MTMount.

teweitsai commented on simulator/README.md:
```diff
@@ -0,0 +1,97 @@
+
+Note:
+
+1. You can only stop the Simulator Component if you stop de TCP Server before.
```
if you stop de TCP Server before --> if you stop "the" TCP Server before.

pcortes80 replied:
Fixed. Thanks.

teweitsai commented on doc/detailsDefault.md:
```diff
@@ -0,0 +1,47 @@
+
+### commands
+
+1. **id** : Specify the name of command that begins with `cmd_`.
```
In addition, I felt the `name` is better than `id` here. But I do understand why you pick the `id` here.

pcortes80 replied:
You are right, I will remove  the `cmd_`.  I picked **id** to be consistence with `README.md` file.

teweitsai reviewed (approved):
This update looks good to me. I left some inline comments of documents. For the code, the following is my review:

[testCreateConfigFilePath.vi]

1. Why do you need the "Clear Errors.vi" in the block diagram? Does it fail in the CTIO Jenkins?

[readConfigFileDefault.vi]

1. Based on your code design, we could get ride of readConfigCmd.vi, openConfigFile.vi, and closeConfigFile.vi. And we can combine the readConfigEvi.vi and readConfigTel.vi to be someething like the readConfigEvtTel.vi and use an enum to decide the event or telemetry.

[openConfigFile.vi]

1. I think this vi can reuse the createConfigFilePath.vi. Or just get ride of this vi.

[readConfigEvt.vi]

1. You can combine this with readConfigTel.vi.

[readConfigTel.vi]

1. 1. You can combine this with readConfigEvt.vi.

pcortes80 commented:
Hi, these are my replies:

[testCreateConfigFilePath.vi]

1. Why do you need the "Clear Errors.vi" in the block diagram? Does it fail in the CTIO Jenkins?

<Reply> This function is not necessary; I just removed this.

[readConfigFileDefault.vi]

1. Based on your code design, we could get ride of readConfigCmd.vi, openConfigFile.vi, and closeConfigFile.vi. And we can combine the readConfigEvi.vi and readConfigTel.vi to be someething like the readConfigEvtTel.vi and use an enum to decide the event or telemetry.

<Reply> 
- removed readConfigCmd.vi.
- removed closeConfigFile.vi.
- removed openConfigFile.vi.
- combined readConfigEvt.vi and readConfigTel.vi into readConfigEvtTel.vi

[openConfigFile.vi]

1. I think this vi can reuse the createConfigFilePath.vi. Or just get ride of this vi.

<Reply> Removed openConfigFile.vi.

[readConfigEvt.vi]

1. You can combine this with readConfigTel.vi.

<Reply> Done.

[readConfigTel.vi]

1. You can combine this with readConfigEvt.vi.

<Reply> Done.
